_rules/auto-update-text-efbfc7.md

```diff
@@ -28,11 +28,7 @@ acknowledgments:
 
 ## Applicability
 
-This rule applies to any [HTML element][] that has a [visible][] [text node][] as a [descendant][] in the [flat tree][], for which all the following is true:
-
-- **changed:** the `innerText` property of the element changes multiple times within a 10 minute time span where there is no [user interaction][]; and
-- **no child changed:** the element does not have [children][child] in the [flat tree][] whose `innerText` property also changes; and
-- **not alone:** the element has an [ancestor][] element in the [flat tree][] with a non-empty `innerText` property whose value is different from the `innerText` of the test target.
+This rule applies to any [HTML element][] that has one or more [visible][] [text nodes][] as [children][] in the [flat tree][], where the combined text from the [text nodes][] changes multiple times within a 10 minute time span where there is no [user interaction][].
```

Comment:
I've been doing a bit of rule archaeology… The "`innerText`" formulation came after [this discussion](https://github.com/act-rules/act-rules.github.io/pull/960#discussion_r345660685), and notably this example:
```html
<div id="x"><b>Hello world</b></div>
<script>
setTimeout(() => document.querySelector('#x').innerHTML = '<i>Goodbye</i>', 1000)
</script>
```
Where the `div` is Applicable (and fails) in the current rule but would be Inapplicable in the new rule (it has no text content), and neither the `b` nor the `i` have text node children that changes (the `i` could as well be a new `b`, different from the existing one, with regular updates).

---
The actual move to use `innerText` seems to be [this commit](https://github.com/act-rules/act-rules.github.io/pull/960/commits/b2ac5cb6550e7199a5484312d5ab07e3deac2588) which was done apparently [after some not archived discussion](https://github.com/act-rules/act-rules.github.io/pull/960#pullrequestreview-323610350).
I have some blurry memories of the fact that even something like all our examples that use `element.innerText = "foo"` to update the text result in **creating** a new text node rather than **updating** the existing one. @carlosapaduarte @WilcoFiers do you remember anything else?

Then of course, the new Applicability uses "combined text" which sounds like a non-technical way to say `innerText`. I'd argue we'd need to define "combined text" more precisely, because in our examples where the text nodes are deleted and replaced by new ones, I'm not sure whether the "combined text from the text nodes" is changing. The `innerText` of the parent is changing, but we need to be unambiguous that "combined text of the text nodes" means "as seen from the parent point of view".

---
In any case, due to the first example in this comment, the change is a big change in Applicability, making the rule much less applicable. I'm not sure this is what we want.

Reply:
I appreciate the rule archaeology, that does help me understand why the applicability was formulated like this. I am going to do some additional poking to see if there is a way to restate it to make it a bit simpler. At the very least (if i can't come up with anything) I would like to add a Note that discusses the context of **no child changed** and **not alone**, such as your comment above on where the **not alone** came from and then for **no child changed** how that localizes the target to the descendant closest to the changed text. 

I figured I would need to define "combined text" in more definite terms. My intention is that it is calculated as the union of the text of all the text nodes of a parent( e.g., so if a parent has 3 text nodes a, b, and c, its combined text would be the union of text from a, b, c). This was my attempt to get rid of some of the tree traversal the **no child changed** condition brings with it. But your first example above using `<b>` or `<i>` invalidates that approach so I will have to go back to the drawing board on that.